packages/test/test-utils/src/testObjectProvider.ts

```diff
@@ -474,10 +591,339 @@ export class TestObjectProvider implements ITestObjectProvider {
+		packageEntries: Iterable<[IFluidCodeDetails, fluidEntryPoint]>,
+		loaderProps?: Partial<ILoaderProps>,
+	) {
+		const logger = createMultiSinkLogger({
+			loggers: [this.logger, loaderProps?.logger],
+		});
+
+		const loader = new this.LoaderConstructorForCreating({
+			...loaderProps,
+			logger,
+			codeLoader: loaderProps?.codeLoader ?? new LocalCodeLoader(packageEntries),
+			urlResolver: loaderProps?.urlResolver ?? this.urlResolver,
+			documentServiceFactory:
+				loaderProps?.documentServiceFactory ?? this.documentServiceFactory,
+		});
+
+		this._loaderContainerTracker.add(loader);
+		return loader;
+	}
+
+	private createLoaderForLoading(
+		packageEntries: Iterable<[IFluidCodeDetails, fluidEntryPoint]>,
+		loaderProps?: Partial<ILoaderProps>,
+	) {
+		const logger = createMultiSinkLogger({
+			loggers: [this.logger, loaderProps?.logger],
+		});
+
+		const loader = new this.LoaderConstructorForLoading({
+			...loaderProps,
+			logger,
+			codeLoader: loaderProps?.codeLoader ?? new LocalCodeLoader(packageEntries),
+			urlResolver: loaderProps?.urlResolver ?? this.urlResolver,
+			documentServiceFactory:
+				loaderProps?.documentServiceFactory ?? this.documentServiceFactory,
+		});
+
+		this._loaderContainerTracker.add(loader);
+		return loader;
+	}
+
+	/**
+	 * {@inheritDoc ITestObjectProvider.createLoader}
+	 */
+	public createLoader(
+		packageEntries: Iterable<[IFluidCodeDetails, fluidEntryPoint]>,
+		loaderProps?: Partial<ILoaderProps>,
+	) {
+		// This will increment `_loadCount` AFTER each create call.
+		return this._loadCount++ % 2 === 0
+			? this.createLoaderForCreating(packageEntries, loaderProps)
+			: this.createLoaderForLoading(packageEntries, loaderProps);
+	}
+
+	/**
+	 * {@inheritDoc ITestObjectProvider.createContainer}
+	 */
+	public async createContainer(entryPoint: fluidEntryPoint, loaderProps?: Partial<ILoaderProps>) {
+		if (this._documentCreated) {
+			throw new Error(
+				"Only one container/document can be created. To load the container/document use loadContainer",
+			);
+		}
+		const loader = this.createLoader([[defaultCodeDetails, entryPoint]], loaderProps);
+		const container = await createAndAttachContainer(
+			defaultCodeDetails,
+			loader,
+			this.driverForCreating.createCreateNewRequest(this.documentId),
+		);
+		this._documentCreated = true;
+		// r11s driver will generate a new ID for the new container.
+		// update the document ID with the actual ID of the attached container.
+		this._documentIdStrategy.update(container.resolvedUrl);
+		return container;
+	}
+
+	/**
+	 * {@inheritDoc ITestObjectProvider.loadContainer}
+	 */
+	public async loadContainer(
+		entryPoint: fluidEntryPoint,
+		loaderProps?: Partial<ILoaderProps>,
+		requestHeader?: IRequestHeader,
+	): Promise<IContainer> {
+		// Keep track of which Loader we are about to use so we can pass the correct driver through
+		const driver = this._loadCount % 2 === 0 ? this.driverForCreating : this.driverForLoading;
```

Comment:
What does it mean to load a container with the driverForCreating?  Maybe this is also pointing at a naming issue.

Reply:
I was just trying to ensure that when we load with 1.X we use 1.X across all API layers and when we load with 2.X we use 2.X across all API layers. In practice I'm not sure if the driver specifically will have a large impact on compatibility, but it is good for completeness either way